morphir/runtime/src/org/finos/morphir/runtime/TypeChecker.scala

```diff
@@ -243,34 +267,41 @@ final class TypeChecker(dists: Distributions) {
   def handleConstructor(tpe: UType, fqn: FQName, context: Context): TypeCheckerResult = {
     import Extractors.Types.*
     val fromChildren = List()
-    val (ret, args)  = Utils.uncurryFunctionType(tpe) // TODO: Interleaved function type w/ aliases.
-    val fromTpe = ret match {
-      case NonNativeRef(name, typeArgs) => dists.lookupTypeSpecification(name) match {
-          case Right(T.Specification.CustomTypeSpecification(typeParams, ctors)) =>
-            val newBindings = typeParams.toList.zip(typeArgs.toList).toMap
-            val missedName = helper(
-              fqn.packagePath != name.packagePath || fqn.modulePath != name.modulePath,
-              new OtherTypeError(s"Constructor $fqn does not match type name $name")
-            )
-            val fromCtor = ctors.toMap.get(fqn.localName) match {
-              case Some(ctorArgs) =>
-                checkList(
-                  args.toList,
-                  ctorArgs.toList.map(_._2).map(Utils.applyBindings(_, newBindings)),
-                  context.withPrefix(s"Comparing $fqn constructor value to looked up type ${Succinct.Type(tpe)}")
+    uncurryFunctionType(tpe, context) match {
+      // if there are uncurrying errors, do not proceed to typecheck the constructor, just return the uncurrying errors
+      case Left(error) => List(error)
+      // otherwise run the constructor and return the errors from that
+      case Right((ret, args)) =>
+        val fromTpe = ret match {
+          case NonNativeRef(name, typeArgs) => dists.lookupTypeSpecification(name) match {
+              case Right(T.Specification.CustomTypeSpecification(typeParams, ctors)) =>
+                val newBindings = typeParams.toList.zip(typeArgs.toList).toMap
+                val missedName = helper(
+                  fqn.packagePath != name.packagePath || fqn.modulePath != name.modulePath,
+                  new OtherTypeError(s"Constructor $fqn does not match type name $name")
```

Comment:
If you want to have a friendlier name you can do `${name.toCamelCase} ($name)` ... or better yet let this be driven bu an EvaluationContext setting